### PR TITLE
[candi][azure] Azure vNet DNS config

### PR DESCRIPTION
**Pull request by borg-z**
## Description

Azure Virtual Network (vNet) allows for the configuration of custom DNS servers. After applying this configuration, all servers and resources within this vNet will use the specified DNS servers.

## Why do we need it, and what problem does it solve?

Ability to configure DNS servers across all nodes in the cluster. This ensures consistent DNS resolution throughout the entire cluster environment.

## What is the expected result?

Once nameservers are specified in the AzureClusterConfiguration, these DNS servers will be used on new servers from the first boot and on existing servers upon reboot.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: fix
summary: Added the ability to configure Node DNS servers via the Azure Cloud Provider.
impact: 
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/azure'...

**Comment by borg-z** [comment]
```
[root@candi-10631578567-1-con-1-27-master-0 ~]# cat /etc/resolv.conf
# Generated by NetworkManager
nameserver 1.1.1.1
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/azure'...

File: candi/cloud-providers/azure/openapi/doc-ru-cluster_configuration.yaml [hunk 1]
```diff
@@ -43,6 +43,12 @@ apiVersions:
           Виртуальная сеть — это виртуальная изолированная часть общедоступной сети Azure. Каждая виртуальная сеть выделена для вашей подписки.
 
           **Важно!** Нужно учитывать, что в случае настройки пиринга, использования VPN или связывания сетей других кластеров, адресное пространство сетей не должно пересекаться.
+      nameservers:
+        description: |
+          Конфигурация DNS серверов для виртуальной сети Azure, которая используется для узлов кластера.
+        properties:
+          addresses:
+            description: Список DNS серверов
```
Comment (z9r5) [inline candi/cloud-providers/azure/openapi/doc-ru-cluster_configuration.yaml:51]:
```suggestion
            description: Список DNS серверов.
```

**Review by z9r5 — APPROVED** [review]

**Comment by z9r5** [comment]
/e2e/abort azure;Standard;containerd;1.27 10634362650 10634362650-1-con-1-27 /sys/deckhouse-oss/install:pr9554 azureuser@40.113.163.223

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
@borg-z Don't forget to destroy failed clusters, please

**Comment by name212** [comment]
@z9r5 we cannot delete canceled clusters in current time, but we will can after merge it https://github.com/deckhouse/deckhouse/pull/9602

**Review by name212 — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]
API - approve

**Review by name212 — APPROVED** [review]

**Pull request merged**